database/migrations/2022_09_20_194258_change_roles_to_new_regulations.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+use App\Models\Role;
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class ChangeRolesToNewRegulations extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::table('roles')->updateOrInsert(
+            ['name'=>'student-council-secretary'],
+            ['has_workshops' => 0, 'has_objects' => 0]
+        );
+        DB::table('roles')->updateOrInsert(
+            ['name'=>'student-council-secretary'],
+            [ 'has_workshops' => 0, 'has_objects' => 0]
+        );
+        DB::table('role_objects')->where('name', 'vice-president')->delete();
+        DB::table('role_objects')->where('name', 'economic-leader')->delete();
+        DB::table('role_objects')->where('name', 'science-leader')->delete();
+        DB::table('role_objects')->where('name', 'economic-member')->delete();
+        DB::table('role_objects')->where('name', 'science-member')->delete();
+        DB::table('role_objects')->updateOrInsert(['role_id'=> Role::firstWhere('name', 'student-council')->id, 'name' => 'science-vice-president'], []);
+        DB::table('role_objects')->updateOrInsert(['role_id'=> Role::firstWhere('name', 'student-council')->id, 'name' => 'economic-vice-president'], []);
+        DB::table('role_objects')->updateOrInsert(['role_id'=> Role::firstWhere('name', 'student-council')->id, 'name' => 'cultural-referent'], []);
+        DB::table('role_objects')->updateOrInsert(['role_id'=> Role::firstWhere('name', 'student-council')->id, 'name' => 'community-referent'], []);
+        DB::table('role_objects')->updateOrInsert(['role_id'=> Role::firstWhere('name', 'student-council')->id, 'name' => 'communication-referent'], []);
+        DB::table('role_objects')->updateOrInsert(['role_id'=> Role::firstWhere('name', 'student-council')->id, 'name' => 'sport-referent'], []);
```

Comment:
Are you sure these work? I think these get matched by the id and the new name. Even if so, you might want to use InsertOrIgnore to be more clear.

Reply:
After a session of googling this was the method that works. insertOrIgnore just ignores that there exists another record like this and adds it regardless.
updateOrInsert works by looking at the first argument if there are matches corresponding to that input, and if there are updates the second argument. If there aren't any, the first and second input will be inserted combined.